db/c.cc

```diff
@@ -996,6 +996,53 @@ void rocksdb_multi_get_cf(
   }
 }
 
+extern ROCKSDB_LIBRARY_API unsigned int rocksdb_key_may_exist(
+    rocksdb_t* db, const rocksdb_readoptions_t* options, const char* key,
+    size_t key_len, char** value, size_t* val_len, const char* timestamp,
+    size_t timestamp_len, unsigned char* value_found) {
+  assert(value != nullptr);
+  assert(val_len != nullptr);
+  std::string tmp;
+  std::string time;
+  if (timestamp) {
+    time.assign(timestamp, timestamp_len);
+  }
+  bool found = false;
+  const bool result = db->rep->KeyMayExist(options->rep, Slice(key, key_len),
+                                           &tmp, timestamp ? &time : nullptr,
+                                           value_found ? &found : nullptr);
+  *val_len = tmp.size();
+  *value = CopyString(tmp);
```

Comment:
Though this function does not return a Status, our convention is not to obligate the caller to free anything (except Status message) unless Status is OK. I think the best analogy here is only to obligate the caller to free `*value` if `*value_found`. Any motivation for this way?

Reply:
I just mirrored the C++ API, but I agree that your proposition makes the C interface more ergonomic. I have updated the pull request.